### PR TITLE
[firtool] Remove many unsued CL options

### PR DESCRIPTION
**Pull request by youngar**
This PR removes many hidden developer options from firtool.  These options were more useful when we would regularly have crashes in specific passes and a developer would want to temporarilly bypass the pass.  Now that we are much more stable, these options don't add any utility, and we risk that a user may find and start using one of these options.

This change also:
  - removes the unused `emit-metadata` flag
  - stops predicating the dedup pass on `--disable-opt`, which allows us to disable optimization and run the dedup pass.
  - restores the original intent of the test in `chirrtl.fir`, which was to make sure that optimizations were not affecting enable inferrence of CHIRRTL read ports.

### CODE REVIEW

File: test/firtool/chirrtl.fir [hunk 1]
```diff
@@ -8,33 +8,32 @@
...
+    ; CHECK: .R1_en (1'h1),
     wire _T_1: UInt<4>
     _T_1 <= addr
     when p:
       read mport testport1 = testmem[_T_1], clock
     out1 <= testport1
-  ; No write to the memory, initialized with invalid, hence optimized away.
 
-  ; CHEC-OLD: assign out0 = 8'h0;
-  ; CHEC-OLD: assign out1 = 8'h0;
```
Comment (seldridge) [inline test/firtool/chirrtl.fir:37]: What was going on here??? 😂

File: tools/firtool/firtool.cpp [hunk 2]
```diff
@@ -329,39 +249,17 @@ static cl::opt<bool> exportModuleHierarchy(
     cl::desc("Export module and instance hierarchy as JSON"), cl::init(false),
     cl::cat(mainCategory));
 
-static cl::opt<bool>
-    disableCheckCombCycles("disable-check-comb-cycles",
-                           cl::desc("Disable the CheckCombCycles pass"),
-                           cl::init(false), cl::Hidden, cl::cat(mainCategory));
```
Comment (seldridge) [inline tools/firtool/firtool.cpp:335]: This type of thing _could_ have a use. However, let's just remove it and we can add it back in later if needed.

File: tools/firtool/firtool.cpp [hunk 3]
```diff
@@ -719,36 +602,31 @@ static LogicalResult processBuffer(
 
   // Run the infer-rw pass, which merges read and write ports of a memory with
   // mutually exclusive enables.
-  if (!disableInferRW)
+  if (!disableOptimization)
     pm.nest<firrtl::CircuitOp>().nest<firrtl::FModuleOp>().addPass(
         firrtl::createInferReadWritePass());
```
Comment (seldridge) [inline tools/firtool/firtool.cpp:607]: I think this should be unconditionally on now. Read-write inference isn't an optimization so much as a defined lowering from CHIRRTL. WDYT?

Comment (youngar) [reply tools/firtool/firtool.cpp:607]: I am not really sure about this one - I believe that the pass is is sort of "best effort" and doesn't really have a spec'ed lowering, and I am pretty sure it is relying on canonicalizations running beforehand to get the best result.  Theoretically  moving this after cross-module optimizations could open up even more port simplifications.

Comment (youngar) [reply tools/firtool/firtool.cpp:607]: I think we might be talking about different passes: this is not the `LowerCHIRRTL` pass, which does have its own "port direction" inference.  This is the one which combines `Read` and `Write` ports into a single `ReadWrite` port.

**Pull request merged**